common/djangoapps/student/helpers.py

```diff
@@ -122,8 +124,16 @@ def check_verify_status_by_course(user, course_enrollment_pairs, all_course_mode
         # verification status.
         if verified_mode is not None and enrollment.mode in CourseMode.VERIFIED_MODES:
             deadline = verified_mode.expiration_datetime
+
             relevant_verification = SoftwareSecurePhotoVerification.verification_for_datetime(deadline, verifications)
 
+            # Picking the max verification datetime on each iteration
+            if relevant_verification is not None:
+                recent_verification_datetime = max(
+                    recent_verification_datetime if recent_verification_datetime is not None
+                    else relevant_verification.expiration_datetime, relevant_verification.expiration_datetime
```

Comment:
Max has issues with datetime and None comparison.
max(datetime.now(),None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can't compare datetime.datetime to NoneType